src/components/new-request/search.vue

```diff
@@ -445,14 +445,20 @@ export default class Search extends Mixins(CommonMixin, NrAffiliationMixin, Sear
 
     // Conditional for Amalgamation Flow.
     if (this.isAmalgamation) {
+      const isXpro = XproMapping.AML.includes(this.getEntityTypeCd)
+
       // named companies
       if (this.getEntityTypeCd && this.isNamedCompany) return true
-      // unknown companies except societies and federal xpro
-      if (this.getEntityTypeCd && !this.isNumberedEntityType && !this.isSociety && !this.isFederal) return true
-      // societies
+      // exclude numbered companies
+      if (this.getEntityTypeCd && this.isNumberedCompany) return false
+      // "unknown" companies except xpro and societies
+      if (this.getEntityTypeCd && !isXpro && !this.isSociety) return true
+      // societies (special case due to FF)
       if (this.getEntityTypeCd && this.isSociety) return true
-      // xpro except federal, once location is selected
-      if (this.getEntityTypeCd && this.getLocation && !this.isFederal) return true
+      // exclude federal
+      if (this.getEntityTypeCd && this.isFederal) return false
+      // xpro, once location is selected
+      if (this.getEntityTypeCd && (this.isLocationCA || this.isLocationIN)) return true
     }
```

Comment:
PS Isn't it great that I can fix amalgamation logic without interfering with the other flows!!!

Reply:
Tested a few test cases. So far looks good